### ISSUE TITLE
How to fine-tune hyperparameters

### ISSUE THREAD

**Issue by Mashaakim** [thuml/Autoformer#35]
Hello!

I'm using Autoformer on my custom dataset, but it seems like an overfitting.
![image](https://user-images.githubusercontent.com/39026184/164514234-10b380a3-76b8-4c15-9e87-25333ed61ed4.png)

I used default parameters, but it looks like it's not a good idea. Which parameters should I fine-tune, and is there any simple method to do it?

Thank you very much in advance.

**Comment by wuhaixu2016** [comment]
Hi, thanks for your interest.

I think you can follow these steps:
(1) Check your dataset size. If the data is not enough for Autoformer training, you need to re-collect the data.
(2) Analyse the periodicity of data. In your case, I find that the period length is around 180. Thus, I think you can try these hyper-parameters: seq_len=180, label_len=90, pred_len=90. It will be very helpful if you provide a complete period to the encoder or decoder.
(3) Enlarge the model. Adding the channels to 1024 can be useful. (d_model=1024)

Hope this is helpful for you.

**Comment by Mashaakim** [comment]
Thank you for your answer! 

I did what you recommended, it became better, but it can be improved.
What about parameters moving_avg, factor, dropout, batch_size, learning_rate? Can they improve the quality?
![image](https://user-images.githubusercontent.com/39026184/164701242-298253c2-76db-4d34-82f9-f4b872507033.png)

Additionally, what if the data doesn't have a period?

**Comment by wuhaixu2016** [comment]
(1) About hyper-parameter
- I am not familiar with your data. Thus, the most convenient way is to tune these hyper-parameters individually and see what happens.
- In my experience, moving_avg and factor can affect. For your data, I think `moving_avg` can be set around 40. As for the `factor`, you can try the choices in {1,3,5,7}

(2) We adopt the deep decomposition architecture for the non-periodic or weak-periodic time series. We conduct experiments and provide showcases about this topic in appendix of our paper.
